ddp-workspace/projects/ddp-dsm-ui/src/app/sharedLearningUpload/components/uploadFile/uploadFile.component.ts

```diff
@@ -59,29 +59,50 @@ export class UploadFileComponent implements OnDestroy {
   }
 
   public upload(): void {
-    if(!this.doNotAllowUpload) {
+    if (!this.doNotAllowUpload) {
       const selectedFiles: FileList = this.inputElement.nativeElement.files;
       const selectedFile: File = selectedFiles.item(0);
       this.updateUploadButtonBy(HttpRequestStatusEnum.IN_PROGRESS);
       const openLoadingDialog = this.openLoadingDialog(selectedFile.name);
       let somaticResultsFile: SomaticResultsFile;
 
+      this.subscription?.unsubscribe();
+
       this.subscription = this.sharedLearningsHTTPService
         .getSignedUrl(selectedFile, this.participantId)
         .pipe(
-          tap(({somaticResultUpload}: SomaticResultSignedUrlResponse) =>
-            somaticResultsFile = somaticResultUpload),
-          mergeMap(({signedUrl}: SomaticResultSignedUrlResponse) =>
-            this.sharedLearningsHTTPService.upload(signedUrl, selectedFile)),
-          finalize(() => openLoadingDialog.close())
-        )
-        .subscribe({
-          next: () => this.handleSuccess(somaticResultsFile),
-          error: (error: any) => this.handleError(error)
-        });
+          tap(signedUrlResponse => {
+            somaticResultsFile = signedUrlResponse.somaticResultUpload;
+            const headers = new HttpHeaders().set(InterceptorSkipHeader,"true");
+            this.sharedLearningsHTTPService.upload(signedUrlResponse.signedUrl, selectedFile, headers).pipe(
+              tap(uploadResponse => this.handleSuccess(somaticResultsFile)),
+              catchError(err => {
+                // if there's an error storing the file, remove the file
+                // and inform the user to retry
+                console.log('Error uploading ' + selectedFile.name + ' to ' + signedUrlResponse.signedUrl + ':' + JSON.stringify(err));
+                this.sharedLearningsHTTPService.delete(somaticResultsFile.somaticDocumentId)
```

Comment:
Use the existing service to delete the upload, thereby making the endless spinning wheel vanish and removing the upload row from the table.